!#bin/bash/
cd /data/OGVFB/eu_plaquenil/affected

#gzip all the vcf files by using.
module load samtools
ls *.vcf | xargs -n1 bgzip
#index the files by using tabix command
ls *.gz | xargs -n1 tabix -p vcf
#merge affected vcf files by using bcftools
bcftools merge 17-2186_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2187_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2189_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2191_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2192_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2193_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2194_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2198_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2199_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2208_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2212_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2217_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2219_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2221_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2224_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2226_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2229_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2238_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2239_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2240_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2243_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2245_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2247_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2248_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2249_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2256_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2257_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2258_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2259_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2260_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2261_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2262_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2263_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2264_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2266_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2267_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2268_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2269_Miseq26_STARG_Mutation_Report1_Filtered.vcf.gz 17-2270_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2273_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2275_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0772_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0773_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0774_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz | bgzip > affected_all.vcf.gz
#same process for unaffected
cd /data/OGVFB/eu_plaquenil/unaffected
#gzip all the vcf files by using.
module load samtools
ls *.vcf | xargs -n1 bgzip
#index the files by using tabix command
ls *.gz | xargs -n1 tabix -p vcf
#merge
bcftools merge 17-2184_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2185_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2188_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2190_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2195_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2196_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2197_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2200_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2201_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2202_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2203_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2204_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2205_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2206_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2207_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2210_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2211_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2213_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2214_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2215_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2216_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2218_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2220_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2222_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2223_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2225_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2227_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2228_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2230_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2231_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2232_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2233_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2234_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2235_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2236_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2241_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2242_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2244_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2250_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2251_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2252_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2253_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2254_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2255_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2265_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2271_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2272_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2274_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2276_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2277_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0771_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0775_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0776_Miseq26_STARG_Mutation_Report1_Filtered.vcf.gz | bgzip > unaffected_all.vcf.gz
#same process for all vcf files
cd /data/OGVFB/eu_plaquenil/all
#gzip all the vcf files by using.
module load samtools
ls *.vcf | xargs -n1 bgzip
#index the files by using tabix command
ls *.gz | xargs -n1 tabix -p vcf
#merge files
bcftools merge 17-2186_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2187_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2189_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2191_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2192_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2193_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2194_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2198_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2199_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2208_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2212_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2217_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2219_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2221_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2224_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2226_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2229_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2238_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2239_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2240_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2243_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2245_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2247_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2248_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2249_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2256_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2257_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2258_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2259_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2260_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2261_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2262_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2263_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2264_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2266_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2267_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2268_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2269_Miseq26_STARG_Mutation_Report1_Filtered.vcf.gz 17-2270_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2273_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2275_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0772_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0773_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0774_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2184_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2185_Miseq10_STARG_Mutation_Report1_Filtered.vcf.gz 17-2188_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2190_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2195_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2196_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2197_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2200_Miseq16_STARG_Mutation_Report1_Filtered.vcf.gz 17-2201_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2202_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2203_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2204_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2205_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2206_Miseq18_STARG_Mutation_Report1_Filtered.vcf.gz 17-2207_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2210_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2211_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2213_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2214_Miseq19_STARG_Mutation_Report1_Filtered.vcf.gz 17-2215_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2216_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2218_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2220_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2222_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2223_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2225_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2227_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2228_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2230_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2231_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2232_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2233_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2234_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2235_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2236_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2241_Stargardt_Mutation_Report1_Filtered.vcf.gz 17-2242_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2244_Miseq22_STARG_Mutation_Report1_Filtered.vcf.gz 17-2250_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2251_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2252_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2253_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2254_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2255_Miseq23_STARG_Mutation_Report1_Filtered.vcf.gz 17-2265_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2271_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2272_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2274_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2276_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 17-2277_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0771_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0775_Miseq24_STARG_Mutation_Report1_Filtered.vcf.gz 18-0776_Miseq26_STARG_Mutation_Report1_Filtered.vcf.gz | bgzip > all.vcf.gz
#index merged files.
ls *.gz | xargs -n1 tabix -p vcf
#David made gemini databases.
#Call genotypes and export the data in tsv files.
sinteractive
module load gemini
module load csvkit
gemini query --header -q "select * from variants WHERE af_exac_all < 0.01 AND (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_unaffected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/unaffected_output.tsv
cd ../affected
gemini query --show-samples --header -q "select * from variants " v2_affected_all.PED_v2_plaquenil.gemini.db | csvlook -t  | less -S
gemini query --show-samples --header -q "select * from variants " v2_affected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/affected_nofilters.tsv
gemini query --show-samples --header -q "select * from variants WHERE (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_affected_all.PED_v2_plaquenil.gemini.db | csvlook -t  | less -S
gemini query --show-samples --header -q "select * from variants WHERE (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_affected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/affected_output_2.tsv
gemini query --header -q "select * from variants WHERE af_exac_all < 0.01 AND (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_affected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/affected_output.tsv
gemini query --show-samples --header -q "select * from variants WHERE af_exac_all < 0.01 AND (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_affected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/affected_output_1.tsv
cd ../unaffected
gemini query --show-samples --header -q "select * from variants " v2_unaffected_all.PED_v2_plaquenil.gemini.db | csvlook -t  | less -S
gemini query --show-samples --header -q "select * from variants " v2_unaffected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/unaffected_nofilters.tsv
gemini query --show-samples --header -q "select * from variants WHERE (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_unaffected_all.PED_v2_plaquenil.gemini.db | csvlook -t  | less -S
gemini query --show-samples --header -q "select * from variants WHERE (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_unaffected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/unaffected_output_2.tsv
gemini query --show-samples --header -q "select * from variants WHERE af_exac_all < 0.01 AND (is_coding=1 OR is_splicing=1 OR impact_severity='HIGH' OR clinvar_sig LIKE '%pathogenic%') " v2_unaffected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/unaffected_output_1.tsv

awk -v OFS="\t" 'NR==FNR {a[$3]=$13"\t"$14"\t"$15"\t"$16;next}{$16=$16 "\t" (a[$3]?a[$3]: "-\t-\t-\t-\t")}1' affected_output.tsv unaffected_output
.tsv > tests.tsv #output contains common variants between both files and appends columns 13-16 from fist file into second.

gemini query --show-samples --format sampledetail --header -q "select * from variants " v2_affected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/affected_nofilters_sample_details.tsv

gemini query --show-samples --format sampledetail --header -q "select * from variants " v2_unaffected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/unaffected_nofilters_sample_details.tsv

gemini query --show-samples --format sampledetail --header -q "select * from variants WHERE af_exac_all < 0.01 " v2_affected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/affected_lessthan1percent_sample_details.tsv

gemini query --show-samples --format sampledetail --header -q "select * from variants WHERE af_exac_all < 0.01 " v2_unaffected_all.PED_v2_plaquenil.gemini.db > /data/OGVFB/eu_plaquenil/merged_vcfs/version2/unaffected_lessthan1percent_sample_details.tsv

####query a database by gene# or region###

gemini query --show-samples --format sampledetail --header -q "select * from variants WHERE gene='ABCA4'" v2_unafected_all.PED_v2_plaquenil | csvlook -t | less -S
gemini query --header --region 6:33290840-33297459 --show-samples -q "select * from variants" EGAD00001002656.GATK.PED_EGAD00001002656.gemini.db > /data/OGVFB/AL6628201_variants.tsv

gemini query --show-samples -q "select *, (gts).(phenotype==1) from variants where gene=='C12orf50'" plaquenil.b37.bwa-mem.hardFilterSNP-INDEL.VEP.GRCh37.PED_plaq.gemini.db >> /data/OGVFB/gemini_queries/plaq_WES_candidate_gene
_var_control.tsv
gemini query --show-samples -q "select *, (gts).(phenotype==2) from variants where gene=='C12orf50'" plaquenil.b37.bwa-mem.hardFilterSNP-INDEL.VEP.GRCh37.PED_plaq.gemini.db >> /data/OGVFB/gemini_queries/plaq_WES_candidate_gene
_var_case.tsv